portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -44,10 +47,19 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
       String email = userService.getCurrentUser().getEmail();
       long timestamp = System.currentTimeMillis();
 
+      Document doc =
+        Document.newBuilder().setContent(body).setType(Document.Type.PLAIN_TEXT).build();
+      LanguageServiceClient languageService = LanguageServiceClient.create();
```

Comment:
Creating the `LanguageServiceClient` for every request might be what's slowing the server down. Try creating the `LanguageServiceClient` in the constructor instead, and re-using it for every request. (And please LMK whether this helps, because if so, Emma and Liyah should do it too.)